docs/build-configs.md

```diff
@@ -0,0 +1,76 @@
+Therefore, the build config has to be defined when running the bundler AND when building the app.
+
+### Runtime (Javascript)
+
+For each build config there is a script in the [package.json](../package.json) to ease the process of starting the packager:
+```bash
+yarn start:<build config name>
+```
+
+For the standard development build config `integreat-test-cms` there is also the shortcut `yarn start`.
+
+To access the values of the build config use [this method](../src/modules/app/constants/buildConfig.js).
+
+### Gradle (Android Build)
+
+For each build config there is a script in the [package.json](../package.json) to ease the process of building and installing the app on android:
+```bash
+yarn android:<build config name>
+```
+
+For the standard development build config `integreat-test-cms` there is also the shortcut `yarn android`.
+
+To build `integreat` and `malte` in **production** mode, use the commands `yarn android:integreat:production` and `yarn android:malte:production`.
+
+### XCode (iOS Build)
+
+To use a build config in XCode, select the corresponding xcscheme and hit run or archive as specified in the [manual builds doc](manual-builds.md#ios).
+Before running the app, make sure to start the packager with the right build config as specified [above](#runtime-javascript)
+as the packager started by XCode per default does not select a build config.
+
+## Technical Information
+
+The concept and technical implementation of build configs is inspired by the library [react-native-config](https://github.com/luggit/react-native-config).
+
+Each build config is a set of javascript files (possibly including common files) that can be found in the [corresponding directory](../build-configs/configs).
+The javascript and gradle build config utilities are reading the environment variable `BUILD_CONFIG_NAME` to select the right build config,
+whereas for XCode xccschemes are used as we don't have a bash command to build and install the app there.
+
+### Javascript
+
+To make the selected build config available in the javascript code, the [buildConfig module](../build-configs/index.js)
+reads the environment variable and returns the corresponding javascript object.
+If the env is not set or not a valid name, an error is thrown. 
+
+To access the values of the build config use [this method](../src/modules/app/constants/buildConfig.js).
+
+### Builds
+
+To make the build config available to XCode and Gradle, [xcconfig files](https://nshipster.com/xcconfig/) (simple `<key> = <value>` syntax) are used as this is not possible with .js files.
+Also, XCode has built in support for xcconfig files (and uses them). [Java properties](https://docs.oracle.com/javase/tutorial/essential/environment/properties.html)
+are using the same syntax (while being more powerful), so Gradle can parse the xcconfig files as java properties.
```

Comment:
I think this is not relevant

Reply:
Yeah kind of, I think it is relevant because we should explain why we create xcconfig files and have java properties afterwards. But does not matter :shrug: